numpy/core/src/umath/string_ufuncs.cpp

```diff
@@ -107,6 +120,29 @@ string_cmp(int len1, const character *str1, int len2, const character *str2)
     return 0;
 }
 
+template <bool rstrip, typename character>
```

Comment:
I like the `rstrip` refactor above, but please remove the unnecessary templating here?
`rstrip` is very specific to a silly string comparison functionality that we should probably just delete.